pkg/flowaggregator/flowaggregator.go

```diff
@@ -60,38 +60,53 @@ var (
 		"destinationNodeName",
 		"destinationClusterIPv4",
 		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
+		"egressNetworkPolicyName",
+		"egressNetworkPolicyNamespace",
 	}
 	aggregationFields = []string{
 		"destinationPodName",
 		"destinationPodNamespace",
 		"destinationNodeName",
+		"destinationClusterIPv4",
+		"destinationServicePortName",
+		"ingressNetworkPolicyName",
+		"ingressNetworkPolicyNamespace",
 	}
 )
 
 type flowAggregator struct {
 	flowCollectorAddr  net.Addr
+	collectingProtocol string
 	collectingProcess  ipfix.IPFIXCollectingProcess
 	aggregationProcess ipfix.IPFIXAggregationProcess
-	exportFrequency    time.Duration
+	exportInterval     time.Duration
 	exportingProcess   ipfix.IPFIXExportingProcess
 	templateID         uint16
 	registry           ipfix.IPFIXRegistry
 	sentFlowKeys       []*ipfixintermediate.FlowKey
```

Comment:
do we need to maintain sentFlowKeys? Are there scenarios where we do not send all flow keys?

Reply:
I create this sentFlowKeys list to track the keys which have been successfully sent in this iteration, then delete them in FlowKeyMap.

Reply:
I was thinking about a scenario that some of the flow keys failed to be send out in this cycle and will try again in next round, so I maintain the sentFlowKeys list to track all successfully sent flow keys.

Reply:
Error scenario is unlikely--how about saving the keys for which there is an error rather than those successfully sent? Use that slice to delete the flow records instead. This will save memory as flowKeyRecord map can be very large.

This code will be refactored with the future release of go-ipfix, when there is the notion of inactive flows.